## 5575: 타임카드
- 오래걸림

문제  
JOI 상사는 직원의 근무시간을 타임 카드로 관리하고있다. 직원들은 전용 장비를 사용하여 타임 카드에 출근 시간을 기록한다. 근무를 마치고 퇴근할 때도 타임 카드에 퇴근 시간을 기록한다. 타임카드에서 사용하는 시간단위는 24 시간제를 사용한다.

보안상의 이유로 직원들의 출근 시간은 7시 이후이다. 또한, 모든 직원은 23시 이전에 퇴근한다. 직원의 퇴근 시간은 항상 출근 시간보다 늦다.

입력으로 JOI 상사의 3 명의 직원 A 씨, B 씨, C 씨의 출근 시간과 퇴근 시간이 주어 졌을 때 각 직원의 근무시간을 계산하는 프로그램을 작성하라.

입력  
입력은 3 행으로 구성된다.

첫 번째 줄에는 A 씨의 출근 시간과 퇴근 시간,

두 번째 줄에는 B 씨의 출근 시간과 퇴근 시간,

세 번째 줄에는 C 씨의 출근 시간과 퇴근 시간이 각각 공백으로 구분되어 있다.

시간은 각각 공백으로 구분된 3 개의 정수로 쓰여져있다.

3 개의 정수 h(7 ≦ h ≦ 22), m(0 ≦ m ≦ 59), s(0 ≦ s ≦ 59)는 h시 m 분 s 초를 나타낸다.

출력  
첫 번째 줄에 A 씨의 근무 시간,

두 번째 줄에 B 씨의 근무 시간,

세 번째 줄에 C 씨의 근무 시간을 출력하라.

근무 시간이 h 시간 m 분 s 초이면 h, m, s의 순으로 공백으로 분리하여 출력하라.

In [ ]:
Ach, Acm, Acs, Ath, Atm, Ats = map(int, input().split())
Bch, Bcm, Bcs, Bth, Btm, Bts = map(int, input().split())
Cch, Ccm, Ccs, Cth, Ctm, Cts = map(int, input().split())

Ah = Ath - Ach
Ah = Ah * 60

Am = Atm - Acm
Am += Ah
Am = Am * 60

As = Ats - Acs
As += Am

Ah = As // 3600
Am = (As - Ah * 3600) // 60
As = As - Ah * 3600 - Am * 60


Bh = Bth - Bch
Bh = Bh * 60

Bm = Btm - Bcm
Bm += Bh
Bm = Bm * 60

Bs = Bts - Bcs
Bs += Bm

Bh = Bs // 3600
Bm = (Bs - Bh * 3600) // 60
Bs = Bs - Bh * 3600 - Bm * 60

Ch = Cth - Cch
Ch = Ch * 60

Cm = Ctm - Ccm
Cm += Ch
Cm = Cm * 60

Cs = Cts - Ccs
Cs += Cm

Ch = Cs // 3600
Cm = (Cs - Ch * 3600) // 60
Cs = Cs - Ch * 3600 - Cm * 60

print(Ah, Am, As)
print(Bh, Bm, Bs)
print(Ch, Cm, Cs)

9 0 0 18 0 0
9 0 1 18 0 0
12 14 52 12 15 30
9 0 0
8 59 59
0 0 38


## 25304: 영수증

문제  
준원이는 저번 주에 살면서 처음으로 코스트코를 가 봤다. 정말 멋졌다. 그런데, 몇 개 담지도 않았는데 수상하게 높은 금액이 나오는 것이다! 준원이는 영수증을 보면서 정확하게 계산된 것이 맞는지 확인해보려 한다.

영수증에 적힌,

구매한 각 물건의 가격과 개수
구매한 물건들의 총 금액
을 보고, 구매한 물건의 가격과 개수로 계산한 총 금액이 영수증에 적힌 총 금액과 일치하는지 검사해보자.

입력  
첫째 줄에는 영수증에 적힌 총 금액
$X$가 주어진다.

둘째 줄에는 영수증에 적힌 구매한 물건의 종류의 수
$N$이 주어진다.

이후
$N$개의 줄에는 각 물건의 가격
$a$와 개수
$b$가 공백을 사이에 두고 주어진다.

출력  
구매한 물건의 가격과 개수로 계산한 총 금액이 영수증에 적힌 총 금액과 일치하면 Yes를 출력한다. 일치하지 않는다면 No를 출력한다.

In [ ]:
price = int(input())
num = int(input())

pur = list()
for i in range(num):
  a, b = map(int, input().split())
  pur.append([a, b])

sum = 0
for i in range(len(pur)):
  sum += pur[i][0] * pur[i][1]

if sum == price:
  print("Yes")
else:
  print("No")

260000
4
20000 5
30000 2
10000 6
5000 8
Yes


## 5532: 방학숙제

문제  
상근이는 초등학교에 다닐 때, 방학 숙제를 남들보다 먼저 미리 하고 남은 기간을 놀았다. 방학 숙제는 수학과 국어 문제 풀기이다.

방학은 총 L일이다. 수학은 총 B페이지, 국어는 총 A페이지를 풀어야 한다. 상근이는 하루에 국어를 최대 C페이지, 수학을 최대 D페이지 풀 수 있다.

상근이가 겨울 방학동안 숙제를 하지 않고 놀 수 있는 최대 날의 수를 구하는 프로그램을 작성하시오.

입력  
한 줄에 하나씩 총 다섯 줄에 걸쳐 L, A, B, C, D가 주어진다. (2 ≤ L ≤ 40, 1 ≤ A, B ≤ 1000, 1 ≤ C, D ≤ 100)

항상 방학 숙제를 방학 기간내에 다 할 수 있는 경우만 입력으로 주어진다.

출력  
첫째 줄에 상근이가 놀 수 있는 날의 최댓값을 출력한다.

In [ ]:
L = int(input())
A = int(input())
B = int(input())
C = int(input())
D = int(input())

# 간단하게 math.ceil(값) 활용하면 if문 필요없음
if A%C == 0:
  a = 0
else:
  a = 1
if B%D == 0:
  b = 0
else:
  b = 1

A = A//C + a
B = B//D + b

H = max(A, B)
print(L- H)

20
25
30
6
8
15


## 2845: 파티가 끝나고 난 후

문제  
파티가 끝나고 나면, 사람들은 누가 파티에 왔는지와 얼마나 많은 사람들이 왔는지를 궁금해한다. 보통 파티는 매우 크게 열리기 때문에, 정확하게 몇 명이 참가했는지 알 수가 없다.

지난주 토요일에 상근이는 자신의 3학년 진학을 기념하면서 매우 성대한 파티를 열었다. 그리고, 상근이는 1m2당 몇 명의 사람이 있었는지 알고있다.

상근이의 파티는 정말 엄청난 규모였기 때문에, 대부분의 신문에도 기사가 실렸다. 상근이는 서로 다른 5개의 신문을 보면서 그 기사에 적혀져있는 참가자의 수를 적었다.

상근이는 자신이 알고있는 참가자의 수가 정확하다고 생각한다. 각 신문 기사에 실려있는 참가자의 수가 몇 명 만큼 잘못되어있는지 구하는 프로그램을 작성하시오.

입력  
첫째 줄에 1m2당 사람의 수 L (1 ≤ L ≤ 10)과 파티가 열렸던 곳의 넓이 P (1 ≤ P ≤ 1000)가 주어진다.

둘째 줄에는 각 기사에 실려있는 참가자의 수가 주어진다. 106보다 작은 양의 정수 5개가 주어진다.

출력  
출력은 첫째 줄에 다섯 개의 숫자를 출력해야 한다. 이 숫자는 상근이가 계산한 참가자의 수와  각 기사에 적혀있는 참가자의 수의 차이이다.

In [ ]:
L, P = map(int, input().split())
a, b, c, d, e = map(int, input().split())

cor = L * P
print(a-cor, b-cor, c-cor, d-cor, e-cor)

5 20
99 101 1000 0 97
-1 1 900 -100 -3


## 10039: 평균 점수

문제  
상현이가 가르치는 아이폰 앱 개발 수업의 수강생은 원섭, 세희, 상근, 숭, 강수이다.

어제 이 수업의 기말고사가 있었고, 상현이는 지금 학생들의 기말고사 시험지를 채점하고 있다. 기말고사 점수가 40점 이상인 학생들은 그 점수 그대로 자신의 성적이 된다. 하지만, 40점 미만인 학생들은 보충학습을 듣는 조건을 수락하면 40점을 받게 된다. 보충학습은 거부할 수 없기 때문에, 40점 미만인 학생들은 항상 40점을 받게 된다.

학생 5명의 점수가 주어졌을 때, 평균 점수를 구하는 프로그램을 작성하시오.

입력  
입력은 총 5줄로 이루어져 있고, 원섭이의 점수, 세희의 점수, 상근이의 점수, 숭이의 점수, 강수의 점수가 순서대로 주어진다.

점수는 모두 0점 이상, 100점 이하인 5의 배수이다. 따라서, 평균 점수는 항상 정수이다.

출력  
첫째 줄에 학생 5명의 평균 점수를 출력한다.

In [ ]:
sum = 0
for i in range(5):
  a= int(input())
  if a < 40:
    a = 40
  sum += a

print(sum//5)

10
65
100
30
95
68


## 11945: 뜨거운 붕어빵
- 거꾸로 할 때 a[::-1]의 도움을 받음  

문제  
고려대학교에 입학한 새내기 호돌이는 안암역을 지나다가 한 붕어빵 장수를 만났어요.

“안녕, 안녕, 안녕하십니까, 아저씨! 붕어빵 두 개 주세요.”

“안녕을 세 번 외쳤으니 붕어빵 세 개!”

붕어빵 두 개의 값을 내고 세 개를 받은 호돌이는 기분이 좋았어요. 호돌이가 붕어빵 하나를 꺼내어 한 입 물었는데…. 너무 뜨거워서 그만 붕어빵을 떨어뜨리고 말았어요ㅠㅠ



붕어빵은 자유 낙하운동을 하면서 땅에 떨어졌는데 신기하게도 좌우가 뒤집힌 모양으로 착지했답니다. 호돌이가 붕어빵을 한 입 물기 전의 모양이 입력으로 주어지면, 땅에 떨어졌을 때에는 어떤 모양일지 출력하세요.

입력  
첫째 줄에는 두 개의 정수 N과 M(0≤N,M≤10)이 주어집니다. 둘째 줄부터 N개의 줄에 걸쳐 붕어빵의 모양이 주어집니다. 각 행에는 공백을 나타내는 ‘0‘ 또는 붕어빵을 나타내는 ‘1’이 총 M개 주어집니다.

출력  
입력으로 주어진 붕어빵이 좌우로 뒤집힌 모양을 출력하세요.

In [ ]:
N, M = map(int, input().split())

fish = list()
for i in range(N):
  a = input()
  a = a[::-1]
  fish.append(a)

for i in range(N):
  print(fish[i])

5 7
0010000
0111010
1111111
0111010
0010000
0000100
0101110
1111111
0101110
0000100


## 3046: R2

문제
두 숫자 R1과 R2가 있을 때, 두 수의 평균 S는 (R1+R2)/2와 같다. 상근이는 정인이 생일 선물로 두 숫자 R1과 R2를 주려고 한다. 생일 파티에서 상근이는 정인이에게 이 두 숫자를 말해주고, 정인이는 이 숫자를 받아 적는다. 그리고 나서 기쁜 마음으로 1년동안 이 숫자를 외우면서 산다.

상근이는 R1과 R2를 엄청난 고민 끝에 정했다. 작년에는 R1과 R2를 까먹어서 아무 숫자나 정해서 주었기 때문에, 올해는 까먹지 않기 위해서 평균 S도 같이 기억하려고 한다.

오늘은 정인이 생일이다. 5분 후에 상근이는 생일 선물로 두 숫자 R1과 R2를 말해주어야 하지만, 안타깝게도 R2를 까먹고 말았다. 하지만 R1과 S는 기억하고 있다!

상근이를 도와 R2가 몇 인지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 두 정수 R1과 S가 주어진다. 두 수는 -1000보다 크거나 같고, 1000보다 작거나 같다.

출력
첫째 줄에 R2를 출력한다.

In [ ]:
R1, S = map(int, input().split())
R2 = S * 2  - R1
print(R2)

4 3
2


## 1547: 공

문제  
세준이는 컵 3개를 탁자 위에 일렬로 엎어놓았다. 컵의 번호는 맨 왼쪽 컵부터 순서대로 1번, 2번 3번이고, 세준이는 이 컵을 이용해서 게임을 하려고 한다.

먼저 1번 컵의 아래에 공을 하나 넣는다. 세준이는 두 컵을 고른 다음, 그 위치를 맞바꾸려고 한다. 예를 들어, 고른 컵이 1번과 2번이라면, 1번 컵이 있던 위치에 2번 컵을 이동시키고, 동시에 2번 컵이 있던 위치에 1번 컵을 이동시켜야 한다. 이때 공은 움직이지 않기 때문에, 공의 위치는 맨 처음 1번 컵이 있던 위치와 같다.

세준이는 컵의 위치를 총 M번 바꿀 것이며, 컵의 위치를 바꾼 방법이 입력으로 주어진다. 위치를 M번 바꾼 이후에 공이 들어있는 컵의 번호를 구하는 프로그램을 작성하시오.

입력  
첫째 줄에 컵의 위치를 바꾼 횟수 M이 주어지며, M은 50보다 작거나 같은 자연수이다. 둘째 줄부터 M개의 줄에는 컵의 위치를 바꾼 방법 X와 Y가 주어지며, X번 컵과 Y번 컵의 위치를 서로 바꾸는 것을 의미한다. X와 Y의 값은 3보다 작거나 같고, X와 Y가 같을 수도 있다.

컵을 이동시킨 후에 공이 컵 바깥에 있는 경우는 없다.

출력  
첫째 줄에 공이 들어있는 컵의 번호를 출력한다. 공이 사라져서 컵 밑에 없는 경우에는 -1을 출력한다.

In [ ]:
N = int(input())

cup = [1,2,3]
for i in range(N):
  a, b = map(int, input().split())
  if cup[0] == a:
    if cup[1] == b:
      cup[0] = b
      cup[1] = a
    elif cup[2] == b:
      cup[0] = b
      cup[2] = a
  elif cup[1] == a:
    if cup[0] == b:
      cup[1] = b
      cup[0] = a
    elif cup[2] == b:
      cup[1] = b
      cup[2] = a
  elif cup[2] == a:
    if cup[0] == b:
      cup[2] = b
      cup[0] = a
    elif cup[1] == b:
      cup[2] = b
      cup[1] = a

print(cup[0])

5
2 3
1 3
2 3
2 1
3 1
[3, 2, 1]


## 2752: 세수정렬

문제   
동규는 세수를 하다가 정렬이 하고 싶어졌다.

정수 세 개를 생각한 뒤에, 이를 오름차순으로 정렬하고 싶어졌다.

정수 세 개가 주어졌을 때, 가장 작은 수, 그 다음 수, 가장 큰 수를 출력하는 프로그램을 작성하시오.

입력  
정수 세 개가 주어진다. 이 수는 1보다 크거나 같고, 1,000,000보다 작거나 같다. 이 수는 모두 다르다.

출력  
제일 작은 수, 그 다음 수, 제일 큰 수를 차례대로 출력한다.

In [ ]:
a, b, c = map(int, input().split())
first = 0
second = 0
third = 0
if a >= b:
  if a >= c:
    first = a
    if b >= c:
      second = b
      third = c
    else:
      second = c
      third = b
  else:
    first = c
    second = a
    third = b
else:
  if b >= c:
    first = b
    if a >= c:
      second = a
      third = c
    else:
      second = c
      third = a
  else:
    first = c
    second = b
    third = a

print(third, second, first)

3 1 2
1 2 3


## 24263: 알고리즘 수업 - 알고리즘의 수행 시간 2  

문제  
오늘도 서준이는 알고리즘의 수행시간 수업 조교를 하고 있다. 아빠가 수업한 내용을 학생들이 잘 이해했는지 문제를 통해서 확인해보자.

입력의 크기 n이 주어지면 MenOfPassion 알고리즘 수행 시간을 예제 출력과 같은 방식으로 출력해보자.

MenOfPassion 알고리즘은 다음과 같다.

MenOfPassion(A[], n) {
    sum <- 0;
    for i <- 1 to n
        sum <- sum + A[i]; # 코드1
    return sum;
}
입력  
첫째 줄에 입력의 크기 n(1 ≤ n ≤ 500,000)이 주어진다.

출력  
첫째 줄에 코드1 의 수행 횟수를 출력한다.

둘째 줄에 코드1의 수행 횟수를 다항식으로 나타내었을 때, 최고차항의 차수를 출력한다. 단, 다항식으로 나타낼 수 없거나 최고차항의 차수가 3보다 크면 4를 출력한다.

In [ ]:
n = int(input())

print(n)
print(1)

7
7
1


## 11720: 숫자의 합

문제  
N개의 숫자가 공백 없이 쓰여있다. 이 숫자를 모두 합해서 출력하는 프로그램을 작성하시오.

입력  
첫째 줄에 숫자의 개수 N (1 ≤ N ≤ 100)이 주어진다. 둘째 줄에 숫자 N개가 공백없이 주어진다.

출력  
입력으로 주어진 숫자 N개의 합을 출력한다.

In [ ]:
n = int(input())
s = input()
sum = 0
for i in range(n):
  sum = sum + int(s[i])
print(sum)

11
10987654321
46


## 1264: 모음의 개수

문제  
영문 문장을 입력받아 모음의 개수를 세는 프로그램을 작성하시오. 모음은 'a', 'e', 'i', 'o', 'u'이며 대문자 또는 소문자이다.

입력  
입력은 여러 개의 테스트 케이스로 이루어져 있으며, 각 줄마다 영어 대소문자, ',', '.', '!', '?', 공백으로 이루어진 문장이 주어진다. 각 줄은 최대 255글자로 이루어져 있다.

입력의 끝에는 한 줄에 '#' 한 글자만이 주어진다.

출력  
각 줄마다 모음의 개수를 세서 출력한다

In [ ]:
string = list()
while 1:
  s = input()
  if s == "#":
    break
  string.append(s)


for i in range(len(string)):
  sum = 0
  for j in string[i]:
    if j == "a" or j == "e" or j == "i" or j == "o" or j == "u" or j == "A" or j == "E" or j == "I" or j == "O" or j == "U":
      sum += 1
  print(sum)

How are you today?
Quite well, thank you, how about yourself?
I live at number twenty four.
#
7
14
9


## 2480: 주사위 세 개

문제  
1에서부터 6까지의 눈을 가진 3개의 주사위를 던져서 다음과 같은 규칙에 따라 상금을 받는 게임이 있다.

같은 눈이 3개가 나오면 10,000원+(같은 눈)×1,000원의 상금을 받게 된다.
같은 눈이 2개만 나오는 경우에는 1,000원+(같은 눈)×100원의 상금을 받게 된다.
모두 다른 눈이 나오는 경우에는 (그 중 가장 큰 눈)×100원의 상금을 받게 된다.  
예를 들어, 3개의 눈 3, 3, 6이 주어지면 상금은 1,000+3×100으로 계산되어 1,300원을 받게 된다. 또 3개의 눈이 2, 2, 2로 주어지면 10,000+2×1,000 으로 계산되어 12,000원을 받게 된다. 3개의 눈이 6, 2, 5로 주어지면 그중 가장 큰 값이 6이므로 6×100으로 계산되어 600원을 상금으로 받게 된다.

3개 주사위의 나온 눈이 주어질 때, 상금을 계산하는 프로그램을 작성 하시오.

입력  
첫째 줄에 3개의 눈이 빈칸을 사이에 두고 각각 주어진다.

출력  
첫째 줄에 게임의 상금을 출력 한다.


In [2]:
a, b, c = map(int, input().split())

if a == b == c:
  price = 10000 + a * 1000
elif a == b or b == c or c == a:
  if a == b or b == c:
    price = 1000 + b * 100
  else:
    price = 1000 + c * 100
else:
  if a > b:
    if a > c:
      price = a * 100
    else:
      price = c * 100
  else:
    if b > c:
      price = b * 100
    else:
      price = c * 100

print(price)


2 2 2
12000


## 10768: 특별한 날


문제  
2월 18일은 올해 CCC에 있어서 특별한 날이다.

사용자로부터 정수인 월과 일을 입력받아 날짜가 2월 18일인지 전인지 후인지를 출력하는 프로그램이다.

만약 날짜가 2월 18일 전이면, "Before"을 출력한다. 만약 날짜가 2월 18일 후면, "After"을 출력한다. 만약 2월 18일이라면 "Special" 을 출력한다.

입력  
입력은 두 개의 정수가 각 줄에 걸쳐 입력된다. 이 정수들은 2015년의 날짜에 포함된다.

첫 번째 줄은 1에서 12사이의 월을 입력한다.

두 번째 줄은 1에서 31사이의 그 달에 들어있는 날짜를 입력한다 .

출력  
마지막 줄에 "Before", "After"나 "Special"을 출력한다

In [5]:
month = int(input())
day = int(input())

if month == 2:
  if day == 18:
    print("Special")
  elif day < 18:
    print("Before")
  else:
    print("After")
elif month < 2:
  print("Before")
else:
  print("After")

8
30
After


## 11948: 과목선택

문제  
JOI는 물리, 화학, 생물, 지구과학, 역사, 지리 총 6 과목의 시험을 봤다. 각 시험의 만점은 100점이다.

JOI는 물리, 화학, 생물, 지구과학 4과목 중에서 3 과목을 선택하고 역사, 지리 2 과목 중에서 한 과목을 선택한다.

시험 점수의 합이 가장 높게 나오도록 과목을 선택할 때, JOI가 선택한 과목의 시험 점수의 합을 구하시오.

입력  
입력은 6행으로 되어있으며, 각 행에 1개의 정수가 주어진다.

1행에는 JOI의 물리 시험의 점수 A가 주어진다.

2행에는 JOI의 화학 시험의 점수 B가 주어진다.

3행에는 JOI의 생물 시험의 점수 C가 주어진다.

4행에는 JOI의 지구과학의 시험 점수 D가 주어진다.

5행에는 JOI의 역사 시험의 점수 E가 주어진다.

6행에는 JOI의 지리 시험의 점수 F가 주어진다.

입력한 정수 A, B, C, D, E, F는 모두 0이상 100이하이다.

출력  
JOI가 선택한 과목의 총 점수를 1행에 출력하시오.

In [7]:
A = int(input())
B = int(input())
C = int(input())
D = int(input())
E = int(input())
F = int(input())

e = 0
if A <= B and A <= C and A <= D:
  e = B + C + D
elif B <= A and B <= C and B <= D:
  e = A + C + D
elif C <= A and C <= B and C <= D:
  e = A + B + D
else:
  e = A + B + C

m = 0
if E >= F:
  m = E
else:
  m = F

print(e + m)

15
21
15
42
15
62
140


## 10156: 과자

문제  
동수는 제과점에 과자를 사러 가는데 현재 가진 돈이 모자랄 경우 부모님께 모자란 돈을 받으려고 한다. 과자 한 개의 가격이 K, 사려고 하는 과자의 개수가 N이고, 현재 가진 돈의 액수를 M이라 할 때 여러분은 동수가 부모님께 받아야 하는 모자란 돈을 계산하려고 한다.

예를 들어, 과자 한 개의 가격이 30원, 사려고 하는 과자의 개수가 4개, 현재 동수가 가진 돈이 100원이라 할 때, 동수가 부모님께 받아야 하는 돈은 20원이다. 과자 한 개의 가격이 250원, 사려고 하는 과자의 개수가 2개, 현재 동수가 가진 돈이 140원이라 할 때, 동수가 부모님께 받아야 하는 돈은 360원이다. 과자 한 개의 가격이 20원, 사려고 하는 과자의 개수가 6개, 현재 동수가 가진 돈이 120원이라 할 때 동수가 부모님께 받아야 하는 돈은 0원이다. 과자 한 개의 가격이 20원, 사려고 하는 과자의 개수가 10개, 현재 동수가 가진 돈이 320원이라 할 때 동수가 부모님께 받아야 하는 돈은 역시 0원이다.

과자 한 개의 가격, 사려고 하는 과자의 개수와 동수가 현재 가진 돈의 액수가 주어질 때 동수가 부모님께 받아야 하는 돈의 액수를 출력하는 프로그램을 작성하시오.

입력  
첫 번째 줄에는 과자 한 개의 가격 K, 사려고 하는 과자의 개수 N, 현재 동수가 가진 돈 M이 각각 공백을 사이에 두고 주어진다. 단, K, N은 1,000 이하의 양의 정수이고, M은 10만 이하의 양의 정수이다. (1 ≤ K, N ≤ 1,000, 1 ≤ M ≤ 100,000이다.)

출력  
첫 줄에 동수가 부모님께 받아야 하는 돈의 액수를 출력한다.

In [10]:
K, N, M = map(int, input().split())

total = K * N - M
if total < 0:
  total = 0
print(total)

20 6 120
0


## 5543: 상근날드

문제  
상근날드에서 가장 잘 팔리는 메뉴는 세트 메뉴이다. 주문할 때, 자신이 원하는 햄버거와 음료를 하나씩 골라, 세트로 구매하면, 가격의 합계에서 50원을 뺀 가격이 세트 메뉴의 가격이 된다.

햄버거는 총 3종류 상덕버거, 중덕버거, 하덕버거가 있고, 음료는 콜라와 사이다 두 종류가 있다.

햄버거와 음료의 가격이 주어졌을 때, 가장 싼 세트 메뉴의 가격을 출력하는 프로그램을 작성하시오.

입력  
입력은 총 다섯 줄이다. 첫째 줄에는 상덕버거, 둘째 줄에는 중덕버거, 셋째 줄에는 하덕버거의 가격이 주어진다. 넷째 줄에는 콜라의 가격, 다섯째 줄에는 사이다의 가격이 주어진다. 모든 가격은 100원 이상, 2000원 이하이다.

출력  
첫째 줄에 가장 싼 세트 메뉴의 가격을 출력한다.

In [15]:
a = int(input())
b = int(input())
c = int(input())

d = int(input())
e = int(input())

ham = 0
if a <= b and a <= c:
  ham = a
elif b <= a and b <= c:
  ham = b
else:
  ham = c

drink = 0
if d <= e:
  drink = d
else:
  drink = e

print(ham + drink - 50)

1999
1999
100
189
100
150


## 11365: !밀비 급일


문제  
당신은 길을 가다가 이상한 쪽지를 발견했다. 그 쪽지에는 암호가 적혀 있었는데, 똑똑한 당신은 암호가 뒤집으면 해독된다는 것을 발견했다.

이 암호를 해독하는 프로그램을 작성하시오.

입력  
한 줄에 하나의 암호가 주어진다. 암호의 길이는 500을 넘지 않는다.

마지막 줄에는 "END"가 주어진다. (END는 해독하지 않는다.)

출력  
각 암호가 해독된 것을 한 줄에 하나씩 출력한다.

In [19]:
arr = list()
n = 0
while 1:
  code = input()
  if code == "END":
    break
  arr.append(code)
  n += 1

for i in range(n):
  print(arr[i][::-1])

!edoc doog a tahW
noitacitsufbo
erafraw enirambus detcirtsernu yraurbeF fo tsrif eht no nigeb ot dnetni eW
lla sees rodroM fo drol eht ,ssertrof sih nihtiw delaecnoC
END
What a good code!
obfustication
We intend to begin on the first of February unrestricted submarine warfare
Concealed within his fortress, the lord of Mordor sees all


## 11943: 파일 옮기기

문제  
두 개의 바구니에 사과와 오렌지가 있다. 첫 번째 바구니에는 사과 A개와 오렌지 B개가 있으며 두 번째 바구니에는 사과 C개와 오렌지 D개가 있다.

당신은 한 바구니에 있는 과일 하나를 집어서 다른 바구니로 옮길 수 있다. 이런 식으로 과일을 옮길 때, 한 바구니에는 사과만 있게 하고 다른 쪽에는 오렌지만 있게 하려고 한다.

앞서 말한 조건을 만족하도록 과일을 옮길 때, 과일을 옮기는 최소 횟수를 구하는 프로그램을 작성하여라.

입력   
첫 번째 줄에는 첫 번째 바구니에 있는 사과와 오렌지의 수 A, B가 주어진다. (0 ≤ A, B ≤ 1,000)

두 번째 줄에는 두 번째 바구니에 있는 사과와 오렌지의 수 C, D가 주어진다. (0 ≤ C, D ≤ 1,000)

출력  
사과와 오렌지를 옮기는 최소 횟수를 출력한다.

In [22]:
A, B = map(int, input().split())
C, D = map(int, input().split())

case1 = A + D
case2 = B + C
if case1 >= case2:
  print(case2)
else:
  print(case1)

5 2
6 2
7


## 10101: 삼각형 외우기

문제  
창영이는 삼각형의 종류를 잘 구분하지 못한다. 따라서 프로그램을 이용해 이를 외우려고 한다.

삼각형의 세 각을 입력받은 다음,

세 각의 크기가 모두 60이면, Equilateral
세 각의 합이 180이고, 두 각이 같은 경우에는 Isosceles
세 각의 합이 180이고, 같은 각이 없는 경우에는 Scalene
세 각의 합이 180이 아닌 경우에는 Error
를 출력하는 프로그램을 작성하시오.

입력  
총 3개의 줄에 걸쳐 삼각형의 각의 크기가 주어진다. 모든 정수는 0보다 크고, 180보다 작다.

출력  
문제의 설명에 따라 Equilateral, Isosceles, Scalene, Error 중 하나를 출력한다.

In [23]:
a = int(input())
b = int(input())
c = int(input())

if a+b+c == 180:
  if a == b == c:
    print("Equilateral")
  elif a == b or b == c or c == a:
    print("Isosceles")
  else:
    print("Scalene")
else:
  print("Error")

60
70
50
Scalene
